src/tscore/Diags.cc

```diff
@@ -215,74 +217,20 @@ Diags::print_va(const char *debug_tag, DiagsLevel diags_level, const SourceLocat
                 va_list ap) const
 {
   ink_release_assert(diags_level < DiagsLevel_Count);
-
-  using ts::LocalBufferWriter;
-  LocalBufferWriter<1024> format_writer;
+  ts::LocalBufferWriter<1024> format_writer;
 
   // Save room for optional newline and terminating NUL bytes.
   format_writer.clip(2);
 
-  //////////////////////
-  // append timestamp //
-  //////////////////////
-  {
-    struct timeval tp = ink_gettimeofday();
-    time_t cur_clock  = (time_t)tp.tv_sec;
-    char timestamp_buf[48];
-    char *buffer = ink_ctime_r(&cur_clock, timestamp_buf);
-
-    int num_bytes_written = snprintf(&(timestamp_buf[19]), (sizeof(timestamp_buf) - 20), ".%03d", (int)(tp.tv_usec / 1000));
-
-    if (num_bytes_written > 0) {
-      format_writer.write('[');
-      format_writer.write(buffer + 4, strlen(buffer + 4));
-      format_writer.write("] ", 2);
-    }
-  }
-
-  size_t timestamp_end_offset = format_writer.size();
-
-  ///////////////////////
-  // add the thread name //
-  ///////////////////////
-  format_writer.print("{thread-name} ");
-
-  //////////////////////////////////
-  // append the diag level prefix //
-  //////////////////////////////////
-
-  format_writer.write(level_name(diags_level), strlen(level_name(diags_level)));
-  format_writer.write(": ", 2);
-
-  /////////////////////////////
-  // append location, if any //
-  /////////////////////////////
-
+  format_writer.print("[{timestamp}] ");
+  auto timestamp_offset = format_writer.size();
+  format_writer.print("{thread-name} {}: ", level_name(diags_level));
   if (location(loc, show_location, diags_level)) {
-    char *lp, buf[256];
-    lp = loc->str(buf, sizeof(buf));
-    if (lp) {
-      format_writer.write('<');
-      format_writer.write(lp, std::min(strlen(lp), sizeof(buf)));
-      format_writer.write("> ", 2);
-    }
-  }
-  //////////////////////////
-  // append debugging tag //
-  //////////////////////////
-
-  if (debug_tag != nullptr) {
-    format_writer.write('(');
-    format_writer.write(debug_tag, strlen(debug_tag));
-    format_writer.write(") ", 2);
+    format_writer.print("<{}> ", *loc);
   }
-  //////////////////////////////////////////////////////
-  // append original format string, ensure there is a //
-  // newline, and NUL terminate                       //
-  //////////////////////////////////////////////////////
+  format_writer.print("{}{}", ts::bwf::OptionalAffix(debug_tag, ") ", "("), format_string);
```

Comment:
I couldn't guess what we are doing here without the original code. This doesn't look straight forward to me.

Reply:
I'm trying to work out the best way do `OptionalAffix`. The suffix is the second argument because it's more commonly used (e.g. a trailing separator). I think Leif was using it in some other code. I can change this back, although as before, I like to have my formatting as compact as possible, the best being all in one format statement because that makes the output clearer.